src/alloc/abort.rs

```diff
@@ -0,0 +1,126 @@
+#![allow(clippy::use_self)]
+
+use liballoc::alloc::handle_alloc_error;
+
+use crate::alloc::{
+    AllocRef,
+    BuildAllocRef,
+    DeallocRef,
+    Global,
+    NonZeroLayout,
+    NonZeroUsize,
+    ReallocRef,
+    System,
+};
+use core::ptr::NonNull;
+
+/// Indicates that the allocator is allowed to be used where we panic on OOM.
+pub trait Abort: AllocRef {}
```

Comment:
Is the bound really needed or just a hint to the user, that `Abort` without `AllocRef` does not make much sense? If the latter, I'd remove the bound and require both traits on the methods to be explicit. It also requires one click less for unexperienced user in the documentation.